src/parser/vp9_parser.cpp

```diff
@@ -73,30 +79,117 @@ rocDecStatus Vp9VideoParser::ParseVideoData(RocdecSourceDataPacket *p_data) {
-    // Decode the picture
-    if ((ret = SendPicForDecode()) != PARSER_OK) {
-        ERR(STR("Failed to decode!"));
-        return ret;
+        if (uncompressed_header_.show_existing_frame) {
+            int disp_idx = dpb_buffer_.virtual_buffer_index[uncompressed_header_.frame_to_show_map_idx];
+            if (disp_idx == INVALID_INDEX) {
+                ERR("Invalid existing frame index to show.");
+                return PARSER_INVALID_ARG;
+            }
+            if (pfn_display_picture_cb_) {
+                disp_idx = dpb_buffer_.frame_store[disp_idx].dec_buf_idx;
+                decode_buffer_pool_[disp_idx].use_status |= kFrameUsedForDisplay;
+                decode_buffer_pool_[disp_idx].pts = curr_pts_;
+                // Insert into output/display picture list
+                if (num_output_pics_ >= dec_buf_pool_size_) {
+                    ERR("Display list size larger than decode buffer pool size!");
+                    return PARSER_OUT_OF_RANGE;
+                } else {
+                    output_pic_list_[num_output_pics_] = disp_idx;
+                    num_output_pics_++;
+                }
+            }
+    #if DBGINFO
+            PrintDpb();
+    #endif // DBGINFO
+        } else {
+            pic_stream_data_ptr_ = pic_data_ptr;
+            pic_stream_data_size_ = frame_sizes_[frame_index];
+            num_slices_ = 1;
+
+            if ((ret = FindFreeInDecBufPool()) != PARSER_OK) {
+                return ret;
+            }
+            if ((ret = FindFreeInDpbAndMark()) != PARSER_OK) {
+                return ret;
+            }
+            if ((ret = SendPicForDecode()) != PARSER_OK) {
+                ERR(STR("Failed to decode!"));
+                return ret;
+            }
+    #if DBGINFO
+            PrintDpb();
+    #endif // DBGINFO
+            // Output decoded pictures from DPB if any are ready
+            if (pfn_display_picture_cb_ && num_output_pics_ > 0) {
+                if ((ret = OutputDecodedPictures(false)) != PARSER_OK) {
+                    return ret;
+                }
+            }
+            UpdateRefFrames();
+            pic_count_++;
+            dpb_buffer_.dec_ref_count[curr_pic_.pic_idx]--;
+            CheckAndUpdateDecStatus();
+        }
+        pic_data_ptr += frame_sizes_[frame_index];
     }
-    pic_count_++;
 
     return PARSER_OK;
 }
+void Vp9VideoParser::CheckSuperframe(const uint8_t *p_stream, uint32_t chunk_data_size) {
+    const uint32_t superframe_marker = 6; // 0b110
+    bool is_superframe = false;
+    int num_frames = 1;
+    int size_index = 0;
+    int size_bytes = 0;
+    uint8_t last_byte = p_stream[chunk_data_size - 1];
+    if ((last_byte >> 5) == superframe_marker) {
+        size_bytes = ((last_byte >> 3) & 0x3) + 1;
+        num_frames = (last_byte & 0x7) + 1;
+        size_index = 2 + num_frames * size_bytes;
+        if (chunk_data_size >= size_index) {
+            uint8_t first_byte = p_stream[chunk_data_size - size_index];
```

Comment:
remove this statement and modify the if to directly use p_stream[chunk_data_size - size_index]

Reply:
Done.